src/main/java/org/terasology/weatherManager/systems/BlockPlacingWeatherSystem.java

```diff
@@ -67,39 +57,41 @@ public void postBegin() {
     }
 
     /**
-     * Places snow blocks on the ground when it is snowing.
-     * The event with the id PLACE_SNOW will only be created when it is snowing
-     * and the other events also correspond with the appropriate weather condition.
+     * Places snow blocks on the ground when it is snowing. The event with the id PLACE_SNOW will only be created when
+     * it is snowing and the other events also correspond with the appropriate weather condition.
+     *
      * @param event The event that means it is time to place snow
      * @param worldEntity The entity that sent the event (assumed to be the player)
      */
     @ReceiveEvent
     public void onPlaceEvent(PeriodicActionTriggeredEvent event, EntityRef worldEntity) {
+        final Vector3f position = new Vector3f();
+        final Vector3i blockPos = new Vector3i();
         if (event.getActionId().equals(PLACE_SNOW)) {
-            for(Client currentPlayer : networkSystem.getPlayers()) {
+            for (Client currentPlayer : networkSystem.getPlayers()) {
                 LocationComponent locComp = currentPlayer.getEntity().getComponent(LocationComponent.class);
-                Vector3i playerPos = new Vector3i(locComp.getWorldPosition());
+                Vector3i playerPos = blockPos.set(locComp.getWorldPosition(position), RoundingMode.FLOOR);
 
                 placeSnow(playerPos);
```

Comment:
This looks like we don't need `playerPos` at all here:
```suggestion
                blockPos.set(locComp.getWorldPosition(position), RoundingMode.FLOOR);
                placeSnow(blockPos);
```